lib/icingadb/icingadb-utility.cpp

```diff
@@ -245,6 +245,12 @@ String IcingaDB::GetLowerCaseTypeNameDB(const ConfigObject::Ptr& obj)
 }
 
 long long IcingaDB::TimestampToMilliseconds(double timestamp) {
+	// In addition to the limit of 2^63 of the Icinga DB PostgreSQL schema,
+	// negative timestamps or years with more than four digits may cause problems.
+	if (timestamp <= 0.0 || timestamp > 2e11 /* 8307 AD */) {
```

Comment:
8307, that looks quite arbitrary to me. I can guess how you ended up with that number, i.e. taking a timestamp from the year 9999 and then round it to one significant digit. But how would you explain in the changelog why the limit is now somewhere in the middle of the year 8307? When limiting it to the year 9999, I'd say this is easier to follow.

Reply:
Changing the PR title to "limit output to four year digits" as a solution is somewhat misleading, that suggests every 4 digit year should still work.

I don't see any technical reason for this limit, it just seems to be an arbitrary value where you happend to like the decimal representation.